### PR TITLE
Improve Android file system read performance

### PR DESCRIPTION
**Pull request by folecr**
Faster file system read performance on Android!
- Raw filesystem read is approximately 40% faster
- Game startup is approximately 12% faster
- Uses Native AssetManager
- ... Instead of custom unzip code
- Also, because the code now does not rely on internal data structures used in `zip_support/ZipUtils.*`, it should be re-entrant.

Tests
- Tested with https://github.com/folecr/SlideShowFSTest
- Test results are average of 3 runs
- Tested on Galaxy Nexus with Android 4.2.2

Test 1 https://github.com/folecr/SlideShowFSTest/tree/master/samples/Cpp/RawFSTest
- Reads image data using CCFileUtils
- Time taken to read data
  - Before : 297.5 millis
  - After : 125 millis
- Approximately 40% faster

Test 2 https://github.com/folecr/SlideShowFSTest/tree/master/samples/Cpp/SlideShowFSTest
- Creates and displays sprites from image data
- Approximately 12% faster to display all sprites

### CODE REVIEW

**Comment by dumganhar** [comment]
``` sh
/home/travis/bin/android-ndk/toolchains/arm-linux-androideabi-4.7/prebuilt/linux-x86_64/bin/../lib/gcc/arm-linux-androideabi/4.7/../../../../arm-linux-androideabi/bin/ld: error: cannot find -landroid
```

Why only `TestJavascript` can not find -landroid? Anything else need to be modified for `TestJavascript`?

File: cocos2dx/platform/android/CCFileUtilsAndroid.cpp [hunk 1]
```diff
@@ -22,16 +22,37 @@ of this software and associated documentation files (the "Software"), to deal
...
+
+#include <stdlib.h>
+
+#define  LOG_TAG    "CCFileUtilsAndroid.cpp"
+#define  LOGD(...)  __android_log_print(ANDROID_LOG_DEBUG,LOG_TAG,__VA_ARGS__)
 
 using namespace std;
 
-NS_CC_BEGIN
+static AAssetManager* s_assetmanager;
```
Comment (minggo) [inline cocos2dx/platform/android/CCFileUtilsAndroid.cpp:37]: It is better to initialize it.

File: cocos2dx/platform/android/CCFileUtilsAndroid.cpp [hunk 2]
```diff
@@ -72,16 +90,26 @@ bool FileUtilsAndroid::isFileExist(const std::string& strFilePath)
...
-        if (s_pZipFile->fileExists(strPath))
+        if (strFilePath.find(_defaultResRootPath) != 0)
         {
-            bFound = true;
-        } 
+            // Didn't find "assets/" at the beginning of the path
+        } else {
+            // Found "assets/" at the beginning of the path and we don't want it
+            s += strlen("assets/");
+        }
```
Comment (minggo) [inline cocos2dx/platform/android/CCFileUtilsAndroid.cpp:101]: Do nothing in `if` sentence. 
Why not change to 

``` c++
if (strFilePath.find(_defaultResRootPath) == 0)
{
      // Found "assets/" at the beginning of the path and we don't want it
      s += strlen("assets/");
}
```

**Comment by minggo** [comment]
Dumdanhar said

> > Why only TestJavascript can not find -landroid? Anything else need to be modified for TestJavascript?

It is because doesn't change AssetsManagerTest android project setting.

**Comment by minggo** [comment]
@folecr 
I fixed compiling error in #2990.
And i will modify codes as you said in email in another pull request.
I will let you know when i send the pull request.

**Comment by folecr** [comment]
Thanks!

**Comment by minggo** [comment]
@folecr 
I send #3008 to modify some codes because of the usage of asset manager.
Would you please review it. 
Thanks.

**Pull request merged**